### ISSUE TITLE
Graceful Restart

### ISSUE THREAD

**Issue by mpass99** [openHPI/poseidon#474]
Related to #465

Configure a graceful restart including the components: Systemd Socket, Poseidon, and Ansible.
This should prevent the Poseidon-deployment-related Sentry issues (See #465).

- Try to exclude the WebSocket connections (as the restart should not wait 30 seconds).

**Comment by mpass99** [comment]
No changes are required to match the issues criterias.

- Systemd Socket: No changes are required as the socket holds the connections until Poseidon can again accept connections.
- Poseidon: No changes are required as the HTTP server shutdown gracefully by default (https://github.com/golang/go/blob/de5b418bea70aaf27de1f47e9b5813940d1e15a4/src/net/http/server.go#L2965-L2985). WebSocket connections are excluded by default. Our current `gracefulShutdownWait` of 15s might be appropriate for this issue (and might be increased in #475).
- Ansible: As stated [here](https://unix.stackexchange.com/questions/676576/whats-the-difference-between-systemctl-stop-and-systemctl-kill) the default (graceful) timeout for systemd is 3 minutes. We could adjust it, but it seems appropriate.

**Comment by MrSerth** [comment]
Indeed; I was able to verify this behaviour on staging. Constantly requesting a page will work without a failure, besides restarting Poseidon in the meantime. What we see, and that's desired, is a short delay (here, shown in Iteration 29):

<img width="1303" alt="Bildschirmfoto 2023-12-05 um 23 00 57" src="https://github.com/openHPI/poseidon/assets/7300329/45250803-ed4e-40e2-9788-bc211f708f67">

Further, I tested with a very slow client, taking multiple seconds for a response (and, of course, terminated Poseidon during that time). Everything worked:

<img width="791" alt="Bildschirmfoto 2023-12-05 um 23 03 18" src="https://github.com/openHPI/poseidon/assets/7300329/6385becb-cf7c-4748-9533-92fd0b708a64">

Regarding your comments:

- Systemd Socket :+1:
- Poseidon: Excluding WebSocket connections for now is fine, I am eagerly waiting for the blue-green-deployment. Then, we can adjust the time separately or find another solution. 
- The systemd timeout is fine as well.

Therefore, I am happily closing this issue.
